pymatviz/_preprocess_data.py

```diff
@@ -69,22 +69,11 @@ def get_df_nest_level(df_in: pd.DataFrame, col: str, max_level: int = 10) -> int
     Args:
         df_in (pd.DataFrame): The DataFrame to check.
         col (str): Name of the column to check.
-        max_level (int): Max nest level to try up to. Defaults to 10.
 
     Returns:
         int: The maximum nest level.
     """
-    current_level = 0
-    while current_level <= max_level:
-        # Check if any element in the column is a list or np.ndarray
-        if df_in[col].map(lambda x: isinstance(x, (list, np.ndarray))).any():
-            current_level += 1
-            # Explode the lists into separate rows
-            df_in = df_in.explode(col)
-        else:
-            return current_level
-
-    return current_level
+    return df_in[col].apply(lambda val: np.array(val).ndim).max()
```

Comment:
[`np.asarray`](https://numpy.org/doc/stable/reference/generated/numpy.asarray.html#numpy.asarray) has the advantage of creating a new array only when needed, i.e. only when it's not already an array and could speed this function up on large dataframes

Reply:
also, given it's a one-liner, do we need to make it a function? it's only used once, right?

Reply:
btw, `df.map` is faster than `df.apply` so i always use the former unless i actually need `apply` (basically never)